game/src/game_core/event.rs

```diff
@@ -177,53 +166,47 @@ impl Event {
             }
             // apply direct resource changes if there are any and the event is not handled above
             _ => {
-                Event::send_popup(&self.popup_message, &sender, self.popup_type, &self.name)?;
+                Event::send_popup(&self.event_id.warning(game_state.lng, self.message_id), &sender, self.popup_type, self.event_id)?;
                 if let Some(resources) = self.resources {
                     if restore {
-                        gamestate.player.resources_change =
-                            gamestate.player.resources_change + resources;
+                        game_state.player.resources_change =
+                            game_state.player.resources_change + resources;
                     } else {
-                        gamestate.player.resources_change =
-                            gamestate.player.resources_change - resources;
+                        game_state.player.resources_change =
+                            game_state.player.resources_change - resources;
                     }
                 }
             }
         }
-        info!("Event triggered (restore: {}): {}", restore, self.name);
+        info!("Event triggered (restore: {}): {:?}", restore, self.event_id);
         Ok(())
     }
 
-    /// Returns the name of the event
-    pub fn get_name(&self) -> String {
-        self.name.clone()
-    }
-
-    /// Deletes due events from the gamestates events vector and adds new events
+    /// Deletes due events from the game states events vector and adds new events
     /// # Arguments
-    /// * `gamestate` - The gamestate which is used to access the events vector
+    /// * `game_state` - The game state which is used to access the events vector
     /// * `context` - The game context which is used to access the current tick
-    pub fn update_events(ctx: &Context, gamestate: &mut GameState) -> RLResult {
-        let lng = gamestate.lng;
+    pub fn update_events(ctx: &Context, game_state: &mut GameState) -> RLResult {
         if ctx.time.ticks() % 20 == 0 {
-            gamestate.events.iter_mut().for_each(|event| {
+            game_state.events.iter_mut().for_each(|event| {
                 event.duration = event.duration.saturating_sub(20);
-                if event.name == "Sandsturm" {}
+                if event.event_id == EventId::Sandstorm {}
```

Comment:
This line seems to be an artifact that's no longer needed

Reply:
I was convinced this is something TO BE IMPLEMENTED

Reply:
Maybe it was but it was never implemented and i do not recall what was to be implemented and there is no comment left here.
This was a project for our university and sadly we do not have the time to improve the game much anymore. I tried porting it to webassembly with good-web-game which worked reasonably well after the project was finished and made a MacOS release but im afraid apart from your major improvements that's the only development the game has seen as the team all has stressful jobs